hikari/impl/interaction_server.py

```diff
@@ -189,7 +206,10 @@ def is_alive(self) -> bool:
         """
         return self._server is not None
 
-    async def _fetch_public_key(self) -> ed25519.VerifierT:
+    async def _fetch_public_key(self) -> signing.VerifyKey:
+        # This is kept inline as pynacl is an optional dependency.
+        from nacl import signing
```

Comment:
This will import it every time we want to use it. Instead, should we try and import it when we load the file and then set a flag to know if it has been imported or not, which we can later use:

```py
try:
    from nacl import signing
except ModuleNotFoundError:
    signing = None
```

And then some checks can be set before its used

Reply:
Mypy wouldn't allow that.  also this isn't hot code, with the current approach signing is imported 1-2 times per bot that's initiated and then run 1+ times

Reply:
Fair enough